test/interactions/interactionTests.ts

```diff
@@ -0,0 +1,94 @@
+///<reference path="../testReference.ts" />
+
+var assert = chai.assert;
+
+describe("Interactions", () => {
+  describe("Interactions", () => {
+    var SVG_WIDTH = 400;
+    var SVG_HEIGHT = 400;
+
+    it("can attach/detach interaction to/from component", () => {
+      var svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+      var component = new Plottable.Component();
+      component.renderTo(svg);
+
+      var clickInteraction = new Plottable.Interactions.Click();
+
+      var callbackCalled = false;
+      var callback = () => callbackCalled = true;
+      clickInteraction.onClick(callback);
+
+      clickInteraction.attachTo(component);
+
+      TestMethods.triggerFakeMouseEvent("mousedown", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+      TestMethods.triggerFakeMouseEvent("mouseup", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+      assert.isTrue(callbackCalled, "callback called on clicking Component (mouse)");
+
+      callbackCalled = false;
+      clickInteraction.detachFrom(component);
+
+      TestMethods.triggerFakeMouseEvent("mousedown", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+      TestMethods.triggerFakeMouseEvent("mouseup", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+      assert.isFalse(callbackCalled, "callback was removed from component and should not be called");
+
+      svg.remove();
+    });
```

Comment:
It looks like he would have to add a dispatcher or something. Would it be enough to spy on `_anchor()` and `_unanchor()`, or `<any>`-cast to check `_componentToListenTo`?